### PR TITLE
[INFRA] set up github action to detect latin phrases

### LINKED ISSUE #635: Setup a no-bad-latin github action
> On related issue: would it make sense to "install" the github action used by the Turing way to check our content for Latin expressions to ensure we don't have to redo this in the future?

probably worth looking into, good idea! :+1: we'd have to clarify what else they are linting / checking, and make sure that all of these steps are (or can be) configured to our liking. 

If you have time and interest to do that, I'd appreciate a short report and potentially a new PR :-)

_Originally posted by @sappelhoff in https://github.com/bids-standard/bids-specification/pull/628#issuecomment-701996073_

Opening this issue with the links to the action and its script.

Github action workflow:
https://github.com/alan-turing-institute/the-turing-way/blob/master/.github/workflows/no-bad-latin.yml

Python script called by action:
https://github.com/alan-turing-institute/the-turing-way/blob/master/tests/no-bad-latin.py

Point to figure out if we decide to use it:
- crediting the person who wrote this code.

### PR DESCRIPTION
**Pull request by Remi-Gau**
This relates to the issue #635 .
fixes #635 

TODO
- [x] import and adapt script 
- [x] test script locally
- [x] credit the turing-way
- [x] test the github action

Adapting the code from the Turing way was actually easier than anticipated.

## script

The way this is set up, this should look through the files in the `src` folder ignoring the files with extensions listed in `get_all_files` and look for lines that contain the expressions listed in `read_and_check_files`.

Comments lines are skipped. 

Files added to the `src` folder in a PR are also checked

## github action
It will trigger the script : 
- on a push to master 
- on a PR to the master branch and will only check files that were modified in `src`

## for reviewers

My python is far from fluent so a cross check would be good. :pray: 

~~Any suggestions on how to smartly test a github action? Should I merge into this onto my own master to give this a go.~~
This seems to be set up directly even in this PR. Never mind.

### CODE REVIEW

**Comment by Remi-Gau** [comment]
FYI: you get plenty of errors when running the script locally but the github action passes because on a PR it will only look for files modified in `src` which is not the case in this PR.

File: tools/no-bad-latin.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import os
```
Comment (effigies) [inline tools/no-bad-latin.py:1]: If we've copied from The Turing Way, the appropriate place to credit (IMO) would be here. Something like:

```suggestion
#!/usr/bin/env python
# 
# Detect Latin abbreviations that can be difficult for screenreaders and non-native English speakers
#
# This script initially adopted from The Turing Way from in October 2020.
# https://github.com/alan-turing-institute/the-turing-way/blob/af98c94/tests/no-bad-latin.py

import os
```

File: tools/no-bad-latin.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+# 
+# Detect Latin abbreviations that can be difficult for screenreaders and non-native English speakers
+#
+# This script initially adopted from The Turing Way from in October 2020.
+# https://github.com/alan-turing-institute/the-turing-way/blob/af98c94/tests/no-bad-latin.py
```
Comment (Remi-Gau) [inline tools/no-bad-latin.py:6]:
```suggestion
# This script initially adopted from The Turing Way from in October 2020.
# doi:10.5281/zenodo.3233853
# https://github.com/alan-turing-institute/the-turing-way/blob/af98c94/tests/no-bad-latin.py
```

File: tools/pull_files.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
+"""
+Script to pull changed files in a Pull Request using a GET resquest to the
+GitHub API.
+"""
```
Comment (Remi-Gau) [inline tools/pull_files.py:4]:
```suggestion
GitHub API.

This script initially adopted from The Turing Way from in October 2020.
doi:10.5281/zenodo.3233853
https://github.com/alan-turing-institute/the-turing-way/blob/af98c94/tests/pull_files.py

"""
```

File: .github/workflows/no-bad-latin.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Give your test a name!
```
Comment (Remi-Gau) [inline .github/workflows/no-bad-latin.yml:1]:
```suggestion

This action initially adopted from The Turing Way from in October 2020.
doi:10.5281/zenodo.3233853
https://github.com/alan-turing-institute/the-turing-way/blob/af98c94/.github/workflows/no-bad-latin.yml

```

File: tools/no-bad-latin.py (hunk 1 above)
Comment (Remi-Gau) [reply tools/no-bad-latin.py:8]: Thanks.

Added the DOI and extended that to the other python file and the github action

File: tools/pull_files.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+Script to pull changed files in a Pull Request using a GET resquest to the
```
Comment (sappelhoff) [inline tools/pull_files.py:2]:
```suggestion
Script to pull changed files in a Pull Request using a GET request to the
```

**Review by sappelhoff — APPROVED** [review]
LGTM!

**Review by effigies — APPROVED** [review]

**Comment by effigies** [comment]
Infrastructure, so merging.

**Pull request merged**